### DISCUSSION TITLE
Library

### DISCUSSION THREAD

**Discussion by Rob-FolkDance** [quodlibet/quodlibet#4185] [General]
I cannot load any library into Quod Libet - why not?

**Comment by afontenot** [comment]
Can you explain in more detail what problem you are having, and provide the steps you are trying to follow to import your library?

It would also be helpful if you provided:
 * OS / Quod Libet version information
 * Information about what your library looks like (what format the music is in, where it's located on your computer, etc)

**Comment by Rob-FolkDance** [comment]
Hi,

Sure. I downloaded the version for Mac (4.4.0.dmg) and the first thing to do is load music into the program. In the library tag in preferences there is an option to Add and when I try to from various folders in my documents, nothing is added. Most of my music are mp3 files and they are all greyed out when I try to add them. I have a few AIFF files and they are not greyed out but they still do not load into the library. All of this music plays just fine on Apple Music or other music programs. Is there some setting in Quod Libet that I do not know about? The docs or FAQs do not say anything about problems with adding a library.

Thanks,

Rob

**Reply by afontenot** [reply]
Okay, so what you're doing is opening the preferences, then going to Library, then clicking "Add"? The reason your files are greyed out is that the library isn't for adding individual tracks - you're supposed to select a single folder (like "Music") that contains all or most of your music files. When you do that and click "Scan Library" Quod Libet should do the rest.

Alternatively, you should be able to add a file directly from the File menu.

I did a quick test and I was able to import music via the Library option, and also import specific files via the file menu. mp3, mpa, aaif files all seem to work for me on macOS 11.6. If you can't get it working, maybe let us know what version of macOS you're using, as well as whether you've got a newer M1 or M2 based Mac, as opposed to one of the older Intel Macs.

**Comment by Rob-FolkDance** [comment]
I guess I was not clear. Yes, I am choosing a folder (and have tried several folders) which contain music when trying to add a library in preferences. Nothing gets imported. Alternatively, I have tried to add a file or a folder (from File on the menu bar) and still nothing gets added. In all cases, at the bottom left of the screen, it just says 0 songs. 

I am on a MacBook Air from early 2015 with macOS Monterey 12.6.1 and Intel core i5.
